src/gun.erl

```diff
@@ -776,31 +779,31 @@ loop(State=#state{parent=Parent, owner=Owner, owner_ref=OwnerRef,
 			Transport:close(Socket),
 			down(State, {error, Reason});
 		{OK, _PreviousSocket, _Data} ->
-			loop(State);
+			?MODULE:loop(State);
 		{Closed, _PreviousSocket} ->
-			loop(State);
+			?MODULE:loop(State);
 		{Error, _PreviousSocket, _} ->
-			loop(State);
+			?MODULE:loop(State);
 		{Passive, Socket} when Passive =:= tcp_passive orelse Passive =:= ssl_passive ->
 			Transport:setopts(Socket, [{active, 1000}]),
-			loop(State);
+			?MODULE:loop(State);
 		keepalive ->
 			ProtoState2 = Protocol:keepalive(ProtoState),
 			before_loop(State#state{protocol_state=ProtoState2});
```

Comment:
what about this one?

Reply:
This PR's intention, is to keep the busy processes alive when old beam got purged after hot-patch.

# For busy processes
`before_loop` doesn't really loop itself, it immediately calls `?MODULE:loop`
the chance of having the old module purged after this line, but before the next `?MODULE:loop` is quite slim.

# when this process is mostly idle
if the old module is to be purged during idling,
chances are the it's killed before the `keeplaive` message is received.
an idling gun process getting killed (and restarted) is not a big deal.

Reply:
fixed it anyway, also explained in detail in commit message.